lib/api_client.ex

```diff
@@ -1,23 +1,10 @@
 defmodule GlassFactoryApi.ApiClient do
   @moduledoc """
-  Client which make external HTTP Request
+  Client which make external HTTP Request.
   """
 
-  alias GlassFactoryApi.HttpAdapter
-
   @access_info Application.get_env(:glass_factory_api, GlassFactoryApi.ApiClient)
```

Comment:
Where do you use this `@access_info`?

Reply:
It's used to generate the headers to make the request to GlassFactory API, [here](https://github.com/Cendrao/glass_factory_api/blob/b44194edb01e82b12ad45e855983b9a4b4b49ab8/lib/api_client.ex#L25).

Reply:
As we discussed offline last week, I strongly recommend following the [library guidelines](https://hexdocs.pm/elixir/master/library-guidelines.html#avoid-application-configuration) when it comes to using app config.

### Advantages

* Injecting app config in the function calls allow developers to have more flexibility (think hitting multiple GF accounts in a single app).
* Application config is global, which limits our options for running tests in parallel (`async: true`), with injected arguments we can run all tests in parallel.

### Disadvantages

* Function arity increases by 1.

We can still avoid the arity increase if we store client state in a process. We initiate a disposable client process with a given configuration, then use its reference to make the requests we need with that shared config, spawning more as needed, one per "config set".

Reply:
Forgot to mention, we can still fallback to env vars when no config is provided at runtime, that way we let people config their clients conveniently and don't compromise developer experience.

Reply:
BTW, I can help pairing up on this 👨‍💻👨‍💻

Reply:
I am creating an issue to apply these changes! And it will be great pairing with you! ❤️